src/onDraw/addRowDisplayToggle.js

```diff
@@ -6,45 +6,58 @@ export default function addRowDisplayToggle() {
         this.rows.classed('chm-hidden', false);
     }
 
+    var rows = this.rows[0];
+    var max_id_level = chart.config.id_cols.length - 2;
+
+    function iterateNest(d, id_level) {
+        return d3
+            .nest()
+            .key(d => d[config.id_cols[id_level]])
+            .rollup(function(rows) {
+                if (id_level + 1 <= max_id_level) {
+                    var obj = iterateNest(rows, id_level + 1);
+                } else {
+                    obj = {};
+                }
+                obj.ids = rows.filter(f => f.nest_level == id_level + 1).map(m => m.id);
+                return obj;
+            })
+            .map(d);
+    }
+
+    var childNest = iterateNest(chart.data.raw, 0);
+
+    chart.data.raw.forEach(function(d, i) {
```

Comment:
Might as well do this onDraw() when defining `id_cols`, to avoid unnecessary iterators.